src/backend/marsha/bbb/serializers.py

```diff
@@ -111,3 +122,135 @@ def get_lti_url(self, obj):
             self.context["request"],
             reverse("bbb:classroom_lti_view", args=[obj.id]),
         )
+
+
+class ClassroomDocumentSerializer(
+    UploadableFileWithExtensionSerializerMixin, serializers.ModelSerializer
+):
+    """A serializer to display a ClassroomDocument resource."""
+
+    class Meta:  # noqa
+        model = ClassroomDocument
+        fields = (
+            "classroom",
+            "filename",
+            "id",
+            "is_default",
+            "upload_state",
+            "uploaded_on",
+            "url",
+        )
+        read_only_fields = (
+            "classroom",
+            "id",
+            "upload_state",
+            "uploaded_on",
+            "url",
+        )
+
+    url = serializers.SerializerMethodField()
+    # Make sure classroom UUID is converted to a string during serialization
+    classroom = serializers.PrimaryKeyRelatedField(
+        read_only=True, pk_field=serializers.CharField()
+    )
+
+    def create(self, validated_data):
+        """Force the classroom field to the classroom of the JWT Token if any.
+
+        Parameters
+        ----------
+        validated_data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        dictionary
+            The "validated_data" dictionary is returned after modification.
+
+        """
+        resource = self.context["request"].resource
+
+        if not validated_data.get("classroom_id") and resource:
+            validated_data["classroom_id"] = resource.id
+
+        classroom = Classroom.objects.get(id=validated_data["classroom_id"])
```

Comment:
I think it's more efficient to directly use the `ClassroomDocument` queryset and doing a count without any get.